### ISSUE TITLE
Interoperability with TASTY

### ISSUE THREAD

**Issue by xeno-by** [scalameta/scalameta#147]
One of the fundamental axioms of scala.meta is AST persistence. Since [TASTY](https://docs.google.com/document/d/1h3KUMxsSSjyze05VecJGQ5H2yh7fNADtIf3chD3_wr0/edit) is the platform for AST persistence, we need to be able to interop with it, i.e. to load TASTY as scala.meta trees and to save scala.meta trees into TASTY.

**Comment by xeno-by** [comment]
On our side, this is blocked by #73 and #144. On TASTY side, this is blocked by http://www.google-melange.com/gsoc/proposal/review/org/google/gsoc2015/vova/5899781826150400.

**Comment by xeno-by** [comment]
In 0.0.4, I've implemented a stopgap solution: when stuff is compiled with scalahost, scala.meta trees are serialized via Java serialization and then persisted into a custom section of TASTY. Once we have working serialization/deserialization of scala.reflect trees into TASTY, we'll have full-blown interop with TASTY, but for the time being our TASTY is going to be understandable only by ourselves.

**Comment by xeno-by** [comment]
Superseded by #435.
